src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.filter(mother =>
+    withChildren
+      ? mother.sex === 'f'
+      && people.filter(child => child.mother === mother.name).length > 0
+      : mother.sex === 'f');
+
+  return +(mothers.map(({ born, died }) => died - born)
```

Comment:
you are doing almost the same 3 times, so would be nice to move it to a separate function

Reply:
Done